test/metabase/actions/test_util.clj

```diff
@@ -103,11 +103,18 @@
   `(do-with-dataset-definition actions-test-data (fn [] ~@body)))
 
 (defmacro with-temp-test-data
-  "Sets the current dataset to a freshly created dataset-definition that gets destroyed at the conclusion of `body`.
-   Use this to test destructive actions that may modify the data."
+  "Sets the current dataset to a freshly created table-definitions that gets destroyed at the conclusion of `body`.
+   Use this to test destructive actions that may modify the data.
+    (with-temp-test-data [[\"product\"
+                           [{:field-name \"name\" :base-type :type/Text}]
+                           [[\"Tesla Model S\"]]]
+                          [\"rating\"
+                           [{:field-name \"score\" :base-type :type/Integer}]
+                           [[5]]]]
+      ...)"
   {:style/indent :defn}
-  [dataset-definition & body]
-  `(do-with-dataset-definition (tx/dataset-definition ~(str (gensym)) ~dataset-definition) (fn [] ~@body)))
+  [table-definitions & body]
```

Comment:
It looks like all the non test usages have a single table definition, is it not worth keeping a simple version, rather than wrapping all the call sites in an extra vector?

Given that you've changed from dataset to table, perhaps you could have `with-temp-test-table`, which delegates to the more general `with-temp-test-tables`?

Reply:
I actually need it for a downstream PR: https://github.com/metabase/metabase/pull/38946/files#diff-6598413869c2c34f4fcc6b36e13f5296ab7a822f446240bb82289fc979fbe1bbR215

> Given that you've changed from dataset to table, perhaps you could have with-temp-test-table, which delegates to the more general with-temp-test-tables?

well, I think the original version of this is already weird to start with, I think it should work with multiple in the first place since it called `test-data`, not `test-table`, and a `dataset` definition should have multiple tables.

Like what we have in `mt/with-actions-test-data` or `mt/dataset-definition`